drivers/rtc/rtc_pcf8563.c

```diff
@@ -0,0 +1,460 @@
+				raw_time, sizeof(raw_time));
+	if (ret) {
+		LOG_ERR("Unable to get time. Err: %i", ret);
+		return ret;
+	}
+
+	/* Check integrity, if the first bit is 1 it is ok */
+	if (raw_time[0] & BIT(7)) {
+		LOG_WRN("Clock integrity failed");
+		return -EIO;
+	}
+
+	/* These masks were retrieved from the datasheet
+	 * https://www.nxp.com/docs/en/data-sheet/PCF8563.pdf
+	 * page 6, section 8.2 Register organization.
+	 * Basically, I clean the unused bits and the bits used
+	 * for other stuff
+	 */
+	dest_time->tm_sec = bcd2bin(raw_time[0] & GENMASK(6, 0));
+
+	/* Get minutes */
+	dest_time->tm_min = bcd2bin(raw_time[1] & GENMASK(6, 0));
+
+	/* Get hours */
+	dest_time->tm_hour = bcd2bin(raw_time[2] & GENMASK(5, 0));
+
+	/* Get days */
+	dest_time->tm_mday = bcd2bin(raw_time[3] & GENMASK(5, 0));
+
+	/* Get weekdays */
+	dest_time->tm_wday = raw_time[4] & GENMASK(2, 0);
+
+	/* Get month */
+	dest_time->tm_mon = bcd2bin(raw_time[5] & GENMASK(4, 0));
+
+	/* Get year */
+	dest_time->tm_year = bcd2bin(raw_time[6]);
+
+	/* Day number not used */
+	dest_time->tm_yday = 0;
+
+	/* DST not used  */
+	dest_time->tm_isdst = 0;
+
+	return 0;
+}
+
+
+
+#ifdef CONFIG_RTC_ALARM
+
+static int pcf8563_alarm_get_supported_fields(const struct device *dev, uint16_t id,
+					      uint16_t *mask)
+{
+	ARG_UNUSED(dev);
+
+	/* This device only has one channel*/
+	if (id != 0) {
+		LOG_ERR("invalid ID %d", id);
+		return -EINVAL;
+	}
+
+	*mask = PCF8563_RTC_ALARM_TIME_MASK;
+
+	return 0;
+}
+
+static int pcf8563_alarm_set_time(const struct device *dev, uint16_t id, uint16_t mask,
+				  const struct rtc_time *timeptr)
+{
+	const struct pcf8563_config *config = dev->config;
+	uint8_t regs[4];
+	int ret;
+
+	if (id != 0) {
+		LOG_ERR("invalid ID %d", id);
+		return -EINVAL;
+	}
+
+	if ((mask & ~(PCF8563_RTC_ALARM_TIME_MASK)) != 0) {
+		LOG_ERR("invalid alarm field mask 0x%04x", mask);
+		return -EINVAL;
+	}
+
+	/*
+	 * The first bit is used as enabled/disabled flag.
+	 * The mask will clean it and also the unused bits
+	 */
+	if ((mask & RTC_ALARM_TIME_MASK_MINUTE) != 0) {
+		regs[0] = bin2bcd(timeptr->tm_min) & GENMASK(6, 0);
+	} else {
+		/* First bit to 1 is alarm disabled */
+		regs[0] = BIT(7);
+	}
+
+	if ((mask & RTC_ALARM_TIME_MASK_HOUR) != 0) {
+		regs[1] = bin2bcd(timeptr->tm_hour) & GENMASK(5, 0);
+	} else {
+		regs[1] = BIT(7);
+	}
+
+	if ((mask & RTC_ALARM_TIME_MASK_MONTHDAY) != 0) {
+		regs[2] = bin2bcd(timeptr->tm_mday) & GENMASK(5, 0);
+	} else {
+		regs[2] = BIT(7);
+	}
+
+	if ((mask & RTC_ALARM_TIME_MASK_WEEKDAY) != 0) {
+		regs[3] = bin2bcd(timeptr->tm_wday) & GENMASK(2, 0);
+	} else {
+		regs[3] = BIT(7);
+	}
+
+	ret = i2c_burst_write_dt(&config->i2c, PCF8563_ALARM_REGISTER, regs, sizeof(regs));
+	if (ret) {
+		LOG_ERR("Error when setting alarm: %i", ret);
+		return ret;
+	}
+
+	/* Dont forget to enable interrupts */
+	i2c_reg_write_byte_dt(&config->i2c, PCF8563_CONTROL2_REGISTER, GENMASK(1, 0));
```

Comment:
Fields in registers are usually defined like
```
#define PCF8563_CONTROL2_REGISTER_TIE (1 << 0)
#define PCF8563_CONTROL2_REGISTER_AIE (1 << 1)
uint8_t mask = PCF8563_CONTROL2_REGISTER_TIE_EN | PCF8563_CONTROL2_REGISTER_AIE_EN;
```

 rather than GENMASK(1,0). Its more readable and flexible :)

Reply:
`GENMASK()` and `BIT()` are preferred over manual bit shifts, but yes - please use definitions for the register fields (you can define them using the `GENMASK()` and `BIT()` macros.

Reply:
Thanks for the comment. I adapted the code and I hope that now it follows the style